protocol/invocation/rpcinvocation.go

```diff
@@ -91,6 +93,8 @@ func (r *RPCInvocation) AttachmentsByKey(key string, defaultValue string) string
 }
 
 func (r *RPCInvocation) SetAttachments(key string, value string) {
+	r.lock.Lock()
```

Comment:
it seems that you just use the lock for writing. if u reading the attachment, u should use lock too.

Reply:
@AlexStocks okay.